nexus/src/app/probe.rs

```diff
@@ -60,9 +60,22 @@ impl super::Nexus {
         let (.., authz_project) =
             project_lookup.lookup_for(authz::Action::CreateChild).await?;
 
+        // resolve NameOrId into authz::IpPool
+        let pool = match &new_probe_params.ip_pool {
+            Some(pool) => Some(
+                self.ip_pool_lookup(opctx, &pool)?
+                    .lookup_for(authz::Action::CreateChild)
+                    .await?
+                    .0,
+            ),
+            None => None,
+        };
```

Comment:
While overall I think this PR is a net improvement already, I might try to roll up this logic further with what's in `resolve_pool_for_allocation` because it is awkward to split up this part of the resolution of the pool from the rest.